frontend/src/assets/js/main.js

```diff
@@ -76,38 +82,43 @@ function connect(username, roomCode, difficulty, hackyObject) {
         elem.id = `collaborator-${message.data.username}`;
         elem.appendChild(document.createTextNode(message.data.username));
         list.appendChild(elem);
+        break;
 
       case "disconnect":
         collaborators = collaborators.filter((collaborator) => {
-          collaborator.id !== message.data.id;
+          return collaborator.id !== message.data.id;
         });
         var list = document.getElementById("collabul");
```

Comment:
Move this above the switch statement and declare it as const. Can remove line 80 with this as well.